### PR TITLE
Integrates Consul with "stage one" of HashiCorp Raft library v2.

### PR DESCRIPTION
**Pull request by slackpad**
This isn't ready to merge until we do the proper vendoring but the integration part is ready for review. All unit tests are passing locally and I've booted a test cluster with this :-)

### CODE REVIEW

**Comment by slackpad** [comment]
The dependent changes have been pulled into the https://github.com/hashicorp/raft/tree/library-v2-stage-one branch over in Raft, which will serve as our stable branch for Consul as other Raft refactoring takes place. This is now vendored so this is ready for review.

**Comment by slackpad** [comment]
Switched to govendor so rebase incoming to fix up the metadata.

File: consul/serf.go [hunk 1]
```diff
@@ -230,18 +231,29 @@ func (s *Server) maybeBootstrap() {
...
 	if len(addrs) < s.config.BootstrapExpect {
 		return
 	}
 
-	// Update the peer set
-	s.logger.Printf("[INFO] consul: Attempting bootstrap with nodes: %v", addrs)
-	if err := s.raft.SetPeers(addrs).Error(); err != nil {
-		s.logger.Printf("[ERR] consul: failed to bootstrap peers: %v", err)
+	// Attempt a live bootstrap!
+	s.logger.Printf("[INFO] consul: found expected number of peers, attempting to bootstrap cluster...")
```
Comment (armon) [inline consul/serf.go]: I think its useful to have it log the IPs it is attempting to bootstrap with as a sanity check

Comment (slackpad) [inline consul/serf.go]: Good call - I'll add them to this log message.

File: consul/server.go [hunk 2]
```diff
@@ -332,97 +329,170 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string, w
...
+		// Raft, we almost never want to run recovery based on the old
+		// peers.json file. We create a peers.info file with a helpful
+		// note about where peers.json went, and use that as a sentinel
+		// to avoid ingesting the old one that first time (if we have to
+		// create the peers.info file because it's not there, we also
+		// blow away any existing peers.json file).
+		peersFile := filepath.Join(path, "peers.json")
+		peersInfoFile := filepath.Join(path, "peers.info")
+		if _, err := os.Stat(peersInfoFile); os.IsNotExist(err) {
+			content := []byte(`
```
Comment (armon) [inline consul/server.go]: Can we move this to a constant at the bottom?

**Comment by dadgar** [comment]
LGTM

**Pull request merged**